lib/maths/CLogNormalMeanPrecConjugate.cc

```diff
@@ -577,10 +577,11 @@ class CLogSampleSquareDeviation : core::CNonCopyable {
         result = 0.0;
         for (std::size_t i = 0u; i < m_Samples.size(); ++i) {
             double residual = m_Samples[i];
-            if (residual <= 0.0) {
+            double n = maths_t::countForUpdate(m_Weights[i]);
+            if (residual <= x || !CMathsFuncs::isFinite(residual) ||
```

Comment:
Was it intentional to change `<= 0.0` to `<= x`?

Reply:
Yes. This is actually was actually a bug I spotted while working on this. It causes us to discard a small number of sample we shouldn't be.

Reply:
Although, related tests have failed, so let me double check this.

Reply:
I got the new condition wrong as well :(. Corrected.